### ISSUE TITLE
[Question]: only results of 0 in micro-avg or accuracy

### ISSUE THREAD

**Issue by ronin2304** [flairNLP/flair#3177]
### Question

**Hi , I have a similar problem to #2938 , where I posted basically the same(not sure if it would be spotted since it was closed by the bot, so I decided to do so again here)  but found no solution > the micro_avg remains at 0:**\By class:
              precision    recall  f1-score   support

         PER     0.0000    0.0000    0.0000   15928.0
         LOC     0.0000    0.0000    0.0000   14578.0
         ORG     0.0000    0.0000    0.0000    6247.0
        ANIM     0.0000    0.0000    0.0000    2341.0
       PLANT     0.0000    0.0000    0.0000    1608.0
         DIS     0.0000    0.0000    0.0000    1057.0
         EVE     0.0000    0.0000    0.0000     847.0
        FOOD     0.0000    0.0000    0.0000     705.0
        TIME     0.0000    0.0000    0.0000     651.0
       MEDIA     0.0000    0.0000    0.0000     538.0
         CEL     0.0000    0.0000    0.0000     244.0
       SUPER     0.0000    0.0000    0.0000     166.0
        VEHI     0.0000    0.0000    0.0000      97.0
        INST     0.0000    0.0000    0.0000      36.0
...
   macro avg     0.0000    0.0000    0.0000   45053.0
weighted avg     0.0000    0.0000    0.0000   45053.0

2023-03-29 20:43:45,981 ----------------------------------------------------------------------------------------------------
{'test_score': 0.0,
 'dev_score_history': [0.0, 0.0],
 'train_loss_history': [2.1314844488807965, 0.9294938301940477],
 'dev_loss_history': [0.8666982650756836, 0.6806944012641907]}

**I have checked the dataset as instructed in #2938 and can see that it is correctly parsed below the statistics:** 
Output exceeds the [size limit](command:workbench.action.openSettings?%5B%22notebook.output.textLineLimit%22%5D). Open the full output data [in a text editor](command:workbench.action.openLargeOutput?7e98f9f7-3a17-4e5d-864f-bbe8b1887557)
Corpus: 109754 train + 15680 dev + 31358 test sentences
Sentence[36]: "Präsident Kartnig selbst , der nach dem Erringen des ersten Meistertitels in der Vereinsgeschichte unbedingt einen Star präsentieren wollte , gab die Order den Fitnesszustand zu ignorieren , ohne dies mit Trainer Ivica Osim abzusprechen ." → ["Kartnig"[/PER](https://file+.vscode-resource.vscode-cdn.net/PER), "Ivica Osim"[/PER](https://file+.vscode-resource.vscode-cdn.net/PER)]
{
    "TRAIN": {
        "dataset": "TRAIN",
        "total_number_of_documents": 109754,
        "number_of_documents_per_class": {
            "LOC": 51012,
            "PER": 55420,
            "ORG": 21863,
            "ANIM": 8019,
            "TIME": 2368,
            "DIS": 3590,
            "PLANT": 5447,
            "CEL": 1010,
            "EVE": 2788,
            "FOOD": 2554,
            "SUPER": 601,
            "VEHI": 364,
            "MEDIA": 1991,
            "BIO": 23,
            "INST": 87,
            "PHY": 23
        },
        "number_of_tokens_per_tag": {},
...
            "avg": 17.662244897959184
        }
    }
}

**I am trying the multinerd dataset a sample: **

Nur	O
mit	O
Hilfe	O
von	O
Aktionen	O
in	O
den	O
neutral	O
gebliebenen	O
Ländern	O
Schweiz	B-LOC
,	O
Niederlande	B-LOC
und	O
Schweden	B-LOC
konnte	O
der	O
Schreckenswinter	O
1918	O
/	O
19	O
überstanden	O
werden	O
.	O

Nach	O
einer	O
Kabinettsumbildung	O
war	O
er	O
zuletzt	O
als	O
Nachfolger	O
von	O
Sunthorn	B-PER
Hongladarom	I-PER
vom	O
10	O
.	O

In	O
den	O
folgenden	O
Jahrzehnten	O
wurde	O
ein	O
Teil	O
der	O
Öleinnahmen	O
dadurch	O
an	O
ärmere	O
arabische	O
Staaten	O
in	O
Asien	B-LOC
und	O
Afrika	B-LOC
weitergegeben	O
.	O

**and here is the code I use**:\

from flair.data import Corpus
from flair.datasets import ColumnCorpus
from flair.embeddings import TransformerWordEmbeddings
from flair.models import SequenceTagger
from flair.trainers import ModelTrainer

columns = {0: "text", 1: "ner"}

# Specify the folder containing the dataset files
data_folder = "/home/ed/Desktop/ner/multinerd_flert"

corpus: Corpus = ColumnCorpus(data_folder, columns, train_file="train.tsv", test_file="test.tsv", dev_file="valid.tsv")

# Choose the pre-trained transformer model for embeddings
embedding = TransformerWordEmbeddings(
    model='xlm-roberta-large',
    layers="-1",
    subtoken_pooling="first",
    fine_tune=True,
    use_context=True,
)

# Define the sequence tagger using the FLERT model
tagger = SequenceTagger(hidden_size=256, embeddings=embedding, tag_dictionary=corpus.make_label_dictionary("ner",add_unk=False), tag_type="ner", tag_format="BIOES",use_crf=False, use_rnn=False, reproject_embeddings=False,)

# Train the model
trainer = ModelTrainer(tagger, corpus)
trainer.train(
    'resources/taggers/ner-english-large',
    learning_rate=5.0e-6,
    mini_batch_size=4,
    max_epochs=10,
    mini_batch_chunk_size=1,
)

Any help is very welcome, thank you !
**Edit:** I canged the tag to "BIO" as well and still the result is the same , also inference only returns the input sequence not any tags

**Comment by helpmefindaname** [comment]
Hi @ronin2304 
judging from your outputs, you have canceled the training after 2 epochs, while the loss was still decreasing?

You need to wait until the loss is low enough that the validation score becomes non-zero.
You could try to speed up the learning process by increasing the learning_rate

**Comment by ronin2304** [comment]
Hi @helpmefindaname , thank you.For this run indeed I did stop after 2 epochs,  I had other runs with 10 epochs after epoch 8 I had  Bad Epochs without decrease of loss. Also learning Rate was initially at 0.1 .I will try with 50 epochs and see what the output is, and feedback. Thanks again.
